### PR TITLE
[MAINTENANCE] Move `convert_to_json_serializable` to top-level utils package

### PR DESCRIPTION
**Pull request by Kilo59**
Move `convert_to_json_serializable` to the top-level utils package.
Nesting it within `core.utils` causes a lot of avoidable import cycles.

### CODE REVIEW

File: great_expectations/core/util.py [hunk 1]
```diff
@@ -1,54 +1,28 @@
...
+    from great_expectations.compatibility import pyspark
 
 # import of private class will be removed when deprecated methods are removed from this module
-from great_expectations.exceptions import InvalidExpectationConfigurationError
-from great_expectations.render import RenderedContent
-from great_expectations.types import SerializableDictDot
-from great_expectations.types.base import SerializableDotDict
 
 # Updated from the stack overflow version below to concatenate lists
 # https://stackoverflow.com/questions/3232943/update-value-of-a-nested-dictionary-of-varying-depth
```
Comment (Kilo59) [inline great_expectations/core/util.py:24]:
```suggestion
# Updated from the stack overflow version below to concatenate lists
# https://stackoverflow.com/questions/3232943/update-value-of-a-nested-dictionary-of-varying-depth
```

Comment (Kilo59) [inline great_expectations/core/util.py:24]:
```suggestion
```

File: great_expectations/core/util.py [hunk 2]
```diff
@@ -1,54 +1,25 @@
...
-
-# Updated from the stack overflow version below to concatenate lists
-# https://stackoverflow.com/questions/3232943/update-value-of-a-nested-dictionary-of-varying-depth
-from great_expectations.util import convert_decimal_to_float
 
 if TYPE_CHECKING:
-    from typing_extensions import TypeAlias
+    from great_expectations.compatibility import pyspark
+
+# import of private class will be removed when deprecated methods are removed from this module
```
Comment (Kilo59) [inline great_expectations/core/util.py:21]:
```suggestion
```

File: tests/core/test_convert_to_json_serializable.py
Comment (Kilo59) [inline tests/core/test_convert_to_json_serializable.py:1]: This test needs to move

File: docs/docusaurus/versioned_docs/version-0.17/guides/expectations/creating_custom_expectations/expect_queried_column_value_frequency_to_meet_threshold.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 from typing import Optional, Union
 
 from great_expectations.core.expectation_configuration import ExpectationConfiguration
-from great_expectations.core.util import convert_to_json_serializable
```
Comment (cdkini) [inline docs/docusaurus/versioned_docs/version-0.17/guides/expectations/creating_custom_expectations/expect_queried_column_value_frequency_to_meet_threshold.py]: I don't think we should be touching these versioned docs

Comment (Kilo59) [reply docs/docusaurus/versioned_docs/version-0.17/guides/expectations/creating_custom_expectations/expect_queried_column_value_frequency_to_meet_threshold.py]: Thanks I wasn't even paying attention to this. I'll revert these.

File: tests/core/test_convert_to_json_serializable.py
Comment (billdirks) [reply tests/core/test_convert_to_json_serializable.py:1]: Can you pull this up a directory to tests/test_convert_to_json_serializable.py?

**Review by billdirks — APPROVED** [review]
This LGTM, except for the doc changes that @cdkini mentioned and the test you noted you should move.

Thanks for doing this!

**Pull request merged**